src/PostgREST/App.hs

```diff
@@ -205,8 +204,36 @@ app dbStructure proc conf apiRequest =
                     then responseLBS s [toHeader contentType, r] (toS body)
                     else responseLBS s [r] ""
 
-        (ActionDelete, TargetIdent _, Nothing) ->
-          case mutateSqlParts of
+        (ActionSingleUpsert, TargetIdent (QualifiedIdentifier tSchema tName), Just (PayloadJSON payload pType pKeys)) ->
```

Comment:
I was confused below because `pKeys` made me think "primary keys" rather than "payload keys." Maybe name it `pjKeys`?